### PR TITLE
Add support for the automation config API

### PR DESCRIPTION
**Pull request by gssbzn**

### CODE REVIEW

File: cloudmanager/automation_config.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+		return nil, resp, err
+	}
+
+	return root, resp, err
+}
+
+var _ AutomationService = new(AutomationServiceOp)
+
+type AutomationConfig struct {
+	AgentVersion       *map[string]interface{}   `json:"agentVersion,omitempty"`
```
Comment (gssbzn) [inline cloudmanager/automation_config.go:81]: Right now I don't have much use for many of the structs here so I'm not going to complicate this PR trying to type everything, right now we only care about `Processes` and `ReplicaSets`

File: cloudmanager/automation_config_test.go [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+package cloudmanager
+
+import (
+	"fmt"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+func TestAutomationConfig_Get(t *testing.T) {
```
Comment (gssbzn) [inline cloudmanager/automation_config_test.go:24]: I'm missing a test for the update but this PR is long enough for now, may still add one if required but I'd rather do it on a separate PR

File: cloudmanager/cloudmanager.go [hunk 3]
```diff
@@ -30,8 +30,10 @@ import (
 const (
 	// Version for client
 	Version = "0.1"
-	// DefaultBaseURL API default base URL
-	DefaultBaseURL = "https://cloud.mongodb.com/api/public/v1.0/"
+	// DefaultBaseURL API default base URL for cloud manager
+	DefaultBaseURL = "https://cloud.mongodb.com" + APIPublicV1Path
```
Comment (gssbzn) [inline cloudmanager/cloudmanager.go:34]: The split was asked on an early PR this just adds it here

**Review by MihaiBojin — APPROVED** [review]
LGTM! 💯  for incremental development and small PRs!
